src/redux/modules/cases.ts

```diff
@@ -0,0 +1,114 @@
+import { action } from "typesafe-actions";
+import { RootAction } from "../create";
+import { Dispatch, AnyAction } from "redux";
+import RestAPIClient from "../../api/RestAPIClient";
+
+// Actions
+export const casesActionCreators = {
+  addCases: (cases: Case[]) => action("cases/ADD_CASES", cases),
+  setCases: (cases: Case[]) => action("cases/SET_CASES", cases),
+  removeCase: (receiptNumber: string) =>
+    action("cases/REMOVE_CASE", receiptNumber),
+  clearCases: () => action("cases/CLEAR_CASES"),
+  toggleDetails: (receiptNumber: string) =>
+    action("cases/TOGGLE_DETAILS", receiptNumber),
+  setCaseType: (type: SnoozeState) => action("cases/SET_CASE_TYPE", type),
+  setIsLoading: (isLoading: boolean) =>
+    action("cases/SET_IS_LOADING", isLoading)
+};
+
+export const loadCases = (
+  type: SnoozeState,
+  lastReceiptNumber?: string
+) => async (dispatch: Dispatch<AnyAction>) => {
+  const { setIsLoading, addCases } = casesActionCreators;
+  dispatch(setIsLoading(true));
+  const response =
+    type === "active"
+      ? await RestAPIClient.cases.getActive(lastReceiptNumber)
+      : await RestAPIClient.cases.getSnoozed(lastReceiptNumber);
+  dispatch(setIsLoading(false));
+
+  if (response.succeeded) {
+    dispatch(addCases(response.payload));
+    return;
+  }
+
+  if (response.responseReceived) {
+    const errorJson = await response.responseError.getJson();
+    console.error(errorJson);
+  } else {
+    console.error(response);
+  }
+};
+
+type ActionCreator = typeof casesActionCreators;
+
+export type CasesAction = ReturnType<ActionCreator[keyof ActionCreator]>;
+
+// Initial state
+export type CasesState = {
+  caselist: Case[];
+  type: SnoozeState;
+  isLoading: boolean;
+};
+
+export const initialState: CasesState = {
+  caselist: [],
+  type: "active",
+  isLoading: false
+};
+
+// Reducer
+export default function reducer(
+  state = initialState,
+  action: RootAction
+): CasesState {
+  switch (action.type) {
+    case "cases/ADD_CASES":
+      return {
+        ...state,
+        caselist: [...state.caselist].concat(action.payload)
```

Comment:
this can be written as `state.caselist.concat(action.payload)` since concat does not change `caselist`


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/concat